src/cpu/registers.rs

```diff
@@ -0,0 +1,280 @@
+            R16::AF => (R8::A, R8::F),
+            R16::BC => (R8::B, R8::C),
+            R16::DE => (R8::D, R8::E),
+            R16::HL => (R8::H, R8::L),
+            R16::SP => return self.sp,
+            R16::PC => return self.pc,
+        };
+
+        let high = self.get(high_register);
+        let low = self.get(low_register);
+
+        let mut word: u16 = 0;
+        word |= (high as u16) << 8;
+        word |= low as u16;
+        word
+    }
+
+    pub fn set16(&mut self, register: R16, word: u16) {
+        let (high, low) = match register {
+            R16::AF => (R8::A, R8::F),
+            R16::BC => (R8::B, R8::C),
+            R16::DE => (R8::D, R8::E),
+            R16::HL => (R8::H, R8::L),
+            R16::SP => {
+                self.sp = word;
+                return;
+            }
+            R16::PC => {
+                self.pc = word;
+                return;
+            }
+        };
+
+        let high_value = (word >> 8) as u8;
+        let low_value = word as u8;
+        self.set(high, high_value);
+        self.set(low, low_value);
+    }
+
+    pub fn set16_low(&mut self, register: R16, byte: u8) {
+        let low_register = match register {
+            R16::AF => R8::F,
+            R16::BC => R8::C,
+            R16::DE => R8::E,
+            R16::HL => R8::L,
+            R16::SP => {
+                self.sp &= 0xFF00;
+                self.sp |= byte as u16;
+                return;
+            }
+            R16::PC => {
+                self.pc &= 0xFF00;
+                self.pc |= byte as u16;
+                return;
+            }
+        };
+
+        self.set(low_register, byte);
+    }
+
+    pub fn set16_high(&mut self, register: R16, byte: u8) {
+        let high_register = match register {
+            R16::AF => R8::A,
+            R16::BC => R8::B,
+            R16::DE => R8::D,
+            R16::HL => R8::H,
+            R16::SP => {
+                self.sp &= 0x00FF;
+                self.sp |= (byte as u16) << 8;
+                return;
+            }
+            R16::PC => {
+                self.pc &= 0x00FF;
+                self.pc |= (byte as u16) << 8;
+                return;
+            }
+        };
+
+        self.set(high_register, byte);
+    }
+
+    pub fn get(&self, register: R8) -> u8 {
+        match register {
+            R8::A => self.a,
+            R8::B => self.b,
+            R8::C => self.c,
+            R8::D => self.d,
+            R8::E => self.e,
+            R8::F => self.f.into(),
+            R8::H => self.h,
+            R8::L => self.l,
+        }
+    }
+
+    pub fn set(&mut self, register: R8, byte: u8) {
+        match register {
+            R8::A => self.a = byte,
+            R8::B => self.b = byte,
+            R8::C => self.c = byte,
+            R8::D => self.d = byte,
+            R8::E => self.e = byte,
+            R8::F => self.f = (byte & 0xF0).into(), // Low bits always 0
```

Comment:
This line was different from the rest so I looked around.  I don't think you need to mask it off the way that From  is implemented.  Since you don't need to, it makes me wonder why you did it?